asm_utils/src/data_parser.rs

```diff
@@ -42,16 +47,23 @@ pub fn extract_data_objects<R: Register, F: FunctionOpKind>(
                     assert!(objects.insert(name.clone(), vec![]).is_none());
                 }
                 (".zero" | ".ascii" | ".asciz" | ".word" | ".byte", args) => {
+                    let l = current_label.unwrap();
+                    if !objects.contains_key(l) {
+                        object_order.push(l.to_string());
+                    }
```

Comment:
I think the following should work:

```
objects.entry(l)
   .or_insert_with(|| {
      object_order.push(l.to_string()); 
      Default::default()
   })
   .extend(extract_data_value(dir.as_str(), args))
```

Basically before initialising, add to `object_order`
A bit dirty because it's surprising to have side effects in `or_insert_with` but a bit better than now?

Reply:
yea def, thanks!
btw, I think we shouldn't do the tuple approach we talked about yesterday, because there would be about 10x key repetitions